### ISSUE TITLE
BERT Throughput Calculation

### ISSUE THREAD

**Issue by ali-ampere** [AmpereComputingAI/ampere_model_library#271]
Currently the throughput includes padding within the same batch:
https://github.com/AmpereComputingAI/ampere_model_library/blob/254ead4f5b295eb8c324724b6878c0021959d2b9/natural_language_processing/extractive_question_answering/bert_large/run_mlperf.py#L105
Those paddings should not be counted as meaningful work since they are not affecting the model output. A better measure of total work would be the total number of input tokens:
```python
torch.count_nonzero(input_tensor["attention_mask"]) # using attention mask because I don't know if 0 could be the id of a special token
```

**Comment by ali-ampere** [comment]
An example (captured on the 1st iteration when running with batch size of 4):
```python
input_tensor["input_ids"] = tensor([
        [  101,  1999,  2268,  2054,  2001,  1996,  2561,  1997,  8624,  3018,
          2013,  5765,  1029,   102,  2005,  1996,  2262,  1516,  2410,  2082,
          2095,  3296, 15413,  2001,  1002,  4229,  1010,  2199,  1010,  2007,
          1037,  2561,  3465,  1997,  5270,  1997,  1002,  5401,  1010,  2199,
          1012,  2927,  2289,  1010,  2945,  2007, 29373,  2917,  1002,  3438,
          1010,  2199,  3477,  2498,  2005,  2037,  2336,  2000,  5463,  1010,
          2164,  2282,  1998,  2604,  1012,  2945,  2007, 29373,  2090,  1002,
          3438,  1010,  2199,  2000,  1002,  3770,  1010,  2199,  3477,  2069,
          1037,  2261,  4595,  6363,  2566,  2095,  1010,  1998,  2945,  7414,
          2090,  1002,  6036,  1010,  2199,  1998,  1002,  8380,  1010,  2199,
          3477,  2053,  2062,  2084,  2184,  1003,  1997,  2037,  3296, 29373,
          1012,  1999,  2268,  1010,  5765,  3253,  8624, 21798,  1002,  4601,
          2549,  2454,  2408,  2035,  5408,  5908,  1025,  1031,  2582,  7526,
          2734,  1033,  1002, 16029,  2454,  2234,  2013, 12148,  5029,  1010,
          1002,  3486,  2454,  2013,  2976,  2490,  1010,  1998,  1002,  4464,
          2454,  2013,  2060,  2648,  2490,  1012,  8624,  2561,  6070,  1003,
          1997,  5765,  1005,  1055,  4681,  2005,  8324,  2493,  1010,  2007,
          4681,  2036,  3024,  2011, 10940,  1006,  1022,  1003,  1007,  1998,
          2147,  1011,  2817,  1006,  1018,  1003,  1007,  1012,   102,     0,
             0,     0,     0,     0,     0,     0],
        [  101,  2054,  7017,  1997,  2561,  3361,  4681,  2005,  8324,  2015,
          2013,  5765,  2001,  1999,  1996,  2433,  1997,  8624,  1029,   102,
          2005,  1996,  2262,  1516,  2410,  2082,  2095,  3296, 15413,  2001,
          1002,  4229,  1010,  2199,  1010,  2007,  1037,  2561,  3465,  1997,
          5270,  1997,  1002,  5401,  1010,  2199,  1012,  2927,  2289,  1010,
          2945,  2007, 29373,  2917,  1002,  3438,  1010,  2199,  3477,  2498,
          2005,  2037,  2336,  2000,  5463,  1010,  2164,  2282,  1998,  2604,
          1012,  2945,  2007, 29373,  2090,  1002,  3438,  1010,  2199,  2000,
          1002,  3770,  1010,  2199,  3477,  2069,  1037,  2261,  4595,  6363,
          2566,  2095,  1010,  1998,  2945,  7414,  2090,  1002,  6036,  1010,
          2199,  1998,  1002,  8380,  1010,  2199,  3477,  2053,  2062,  2084,
          2184,  1003,  1997,  2037,  3296, 29373,  1012,  1999,  2268,  1010,
          5765,  3253,  8624, 21798,  1002,  4601,  2549,  2454,  2408,  2035,
          5408,  5908,  1025,  1031,  2582,  7526,  2734,  1033,  1002, 16029,
          2454,  2234,  2013, 12148,  5029,  1010,  1002,  3486,  2454,  2013,
          2976,  2490,  1010,  1998,  1002,  4464,  2454,  2013,  2060,  2648,
          2490,  1012,  8624,  2561,  6070,  1003,  1997,  5765,  1005,  1055,
          4681,  2005,  8324,  2493,  1010,  2007,  4681,  2036,  3024,  2011,
         10940,  1006,  1022,  1003,  1007,  1998,  2147,  1011,  2817,  1006,
          1018,  1003,  1007,  1012,   102,     0],
        [  101,  2044,  2289,  2129,  2172,  2079,  3076,  2013,  2945,  7414,
          2625,  2084,  1002,  3438,  1010,  2199,  3477,  2005,  2082,  1029,
           102,  2005,  1996,  2262,  1516,  2410,  2082,  2095,  3296, 15413,
          2001,  1002,  4229,  1010,  2199,  1010,  2007,  1037,  2561,  3465,
          1997,  5270,  1997,  1002,  5401,  1010,  2199,  1012,  2927,  2289,
          1010,  2945,  2007, 29373,  2917,  1002,  3438,  1010,  2199,  3477,
          2498,  2005,  2037,  2336,  2000,  5463,  1010,  2164,  2282,  1998,
          2604,  1012,  2945,  2007, 29373,  2090,  1002,  3438,  1010,  2199,
          2000,  1002,  3770,  1010,  2199,  3477,  2069,  1037,  2261,  4595,
          6363,  2566,  2095,  1010,  1998,  2945,  7414,  2090,  1002,  6036,
          1010,  2199,  1998,  1002,  8380,  1010,  2199,  3477,  2053,  2062,
          2084,  2184,  1003,  1997,  2037,  3296, 29373,  1012,  1999,  2268,
          1010,  5765,  3253,  8624, 21798,  1002,  4601,  2549,  2454,  2408,
          2035,  5408,  5908,  1025,  1031,  2582,  7526,  2734,  1033,  1002,
         16029,  2454,  2234,  2013, 12148,  5029,  1010,  1002,  3486,  2454,
          2013,  2976,  2490,  1010,  1998,  1002,  4464,  2454,  2013,  2060,
          2648,  2490,  1012,  8624,  2561,  6070,  1003,  1997,  5765,  1005,
          1055,  4681,  2005,  8324,  2493,  1010,  2007,  4681,  2036,  3024,
          2011, 10940,  1006,  1022,  1003,  1007,  1998,  2147,  1011,  2817,
          1006,  1018,  1003,  1007,  1012,   102],
        [  101,  2054,  2003, 15413,  2005,  2262,  1011,  2410,  2095,  2012,
          5765,  1029,   102,  2005,  1996,  2262,  1516,  2410,  2082,  2095,
          3296, 15413,  2001,  1002,  4229,  1010,  2199,  1010,  2007,  1037,
          2561,  3465,  1997,  5270,  1997,  1002,  5401,  1010,  2199,  1012,
          2927,  2289,  1010,  2945,  2007, 29373,  2917,  1002,  3438,  1010,
          2199,  3477,  2498,  2005,  2037,  2336,  2000,  5463,  1010,  2164,
          2282,  1998,  2604,  1012,  2945,  2007, 29373,  2090,  1002,  3438,
          1010,  2199,  2000,  1002,  3770,  1010,  2199,  3477,  2069,  1037,
          2261,  4595,  6363,  2566,  2095,  1010,  1998,  2945,  7414,  2090,
          1002,  6036,  1010,  2199,  1998,  1002,  8380,  1010,  2199,  3477,
          2053,  2062,  2084,  2184,  1003,  1997,  2037,  3296, 29373,  1012,
          1999,  2268,  1010,  5765,  3253,  8624, 21798,  1002,  4601,  2549,
          2454,  2408,  2035,  5408,  5908,  1025,  1031,  2582,  7526,  2734,
          1033,  1002, 16029,  2454,  2234,  2013, 12148,  5029,  1010,  1002,
          3486,  2454,  2013,  2976,  2490,  1010,  1998,  1002,  4464,  2454,
          2013,  2060,  2648,  2490,  1012,  8624,  2561,  6070,  1003,  1997,
          5765,  1005,  1055,  4681,  2005,  8324,  2493,  1010,  2007,  4681,
          2036,  3024,  2011, 10940,  1006,  1022,  1003,  1007,  1998,  2147,
          1011,  2817,  1006,  1018,  1003,  1007,  1012,   102,     0,     0,
             0,     0,     0,     0,     0,     0]])
```
With the old formula, the total amount of work is 784; but there are only 768 meaningful tokens if we remove the paddings.
